### ISSUE TITLE
System hang with display power mgmt set

### ISSUE THREAD

**Issue by gpsamu** [novena-next/docs#26]
Hello, 

Running Buster with kernel 4.4.  I've enabled power management for the display via the "Power management" xfce widget.  When the power management on the display is activated after the specified period of inactivity the built in screen goes black and displays the messages:

```
[365.911055] it6251 2-005c: error -11 writing to edp addr 0x5 
[365.911063] dummy 2-005e: it6251.c:1t6251_init:285 error -1 writing 255 to 5
[365.919987] dummy 2-005e: error -11 writing to lvds addr 0x5 
[365.919993] dummy 2-005e: it6251.c:it6251 init:295 error -1 writing 255 to 5
[366.022805] it6251 2-005c: System status: 0x3e [366.024196] it6251 2-005C: RPCLKCnt: 1520
[ 366.027010] it6251 2-005c: RPC Req: Ox24
[366.025597] it6251 2-005c: Clock: Ox198 
[366.026306] 1t6251 2-005C: Ref Link State: Ox00
[366.028416] 1t6251 2-005c: hactive: 1920 
[366.029818] it6251 2-005c: vactive: 1080
```
At this state the system appears to be hung.  It does not respond to mouse or keyboard input and the display doesn't wake up.  For now, I'm using the screensaver instead and blanking the screen after the specified interval. Wondering if anybody else has seen this or if that power management functionality simply isn't supported on Novena?

**Comment by Jookia** [comment]
Hmm. This shouldn't hang. Does it reply to pings or serial out?

**Comment by gpsamu** [comment]
It does not reply to pings if I recall correctly.  I usually tried to ssh in when this happens, with no luck. Unfortunately I don't have the necessary serial cable.  I've ordered one but it hasn't arrived yet.

**Comment by Jookia** [comment]
Hmm. I think I may hit an issue similiar to this, but I haven't been able to reproduce it as I don't 
Interesting! I've never had a Novena with a battery, but I have hit an issue like this repeatedly but rarely where the display would go blank and won't respond to serial, pings whatever.

I'm honestly not sure what's happening here, but logically it would make sense if this were some kind of power issue that's knocking out the CPU somehow. I haven't scoped this yet since it's a very but finding a more reproducible way to hit this when connected to AC would be very, VERY helpful.

**Comment by gpsamu** [comment]
It *seems* easy to reproduce with the power management enabled for the display. I will disable the screensaver and re-enable the display power mgmt later this evening to see if I can get it to hang again.

**Comment by Jookia** [comment]
Are you on a laptop, desktop or single board?

**Comment by gpsamu** [comment]
It's a desktop model.  So it has the attached display but no battery or battery board.

**Comment by Jookia** [comment]
Oh great, I have that. If you can reliably reproduce it could you write some instructions and I'll try to replicate it?

**Comment by gpsamu** [comment]
Sure, not a problem.  I will try to get to it this evening.  Otherwise tomorrow AM.

**Comment by Jookia** [comment]
No hurry, I'm not exactly actively working on the Novena at the moment.
Partly because of this exact issue among others.

**Comment by gpsamu** [comment]
Scanning the Kosagi forum I came across this: https://www.kosagi.com/forums/viewtopic.php?id=464

My display works fine though.

**Comment by Jookia** [comment]
I've had this happen on the bare board without the eDP adapter board.
Assuming it's the same bug.

**Comment by gpsamu** [comment]
Ok, I was wrong about not being able to ping the system. I had the display power settings configured as followed (see screenshot). 
![PXL_20210116_035942518](https://user-images.githubusercontent.com/16669527/104796736-f99d4580-5786-11eb-8622-a6f3cdbc7104.jpg)

When the display went blank, it displayed the following: 
...
[  503.446974] it6251 2-005c: error -11 writing to edp addr 0x5
[  503.446985] dummy 2-005e: it6251.c:it6251_init:285 error -1 writing 255 to 5
[  503.456541] dummy 2-005e: error -11 writing to lvds addr 0x5
[  503.456553] dummy 2-005e: it6251.c:it6251_init:295 error -1 writing 255 to 5
[  503.550632] it6251 2-005c: System status: 0x3e
[  503.552120] it6251 2-005c: RPCLKCnt: 1520
[  503.553609] it6251 2-005c: Clock: 0x197
[  503.554356] it6251 2-005c: Ref Link State: 0x00
[  503.555111] it6251 2-005c: RPC Req: 0x24
[  503.556599] it6251 2-005c: hactive: 1920
[  503.558083] it6251 2-005c: vactive: 1080
```
However mouse and keyboard input did not wake up the display.  The only thing I could do was ssh to the system and issue a reboot.  

Note that dmesg showed me the following: 

...
[  503.429746] imx-ldb 2000000.aips-bus:ldb@020e0008: 0:imx_ldb_set_clock: LDB_CTRL: 0x000001fd
[  503.429760] imx-ldb 2000000.aips-bus:ldb@020e0008: 0:imx_ldb_set_clock: now: 475200000 want: 484750000
[  503.429785] imx-ldb 2000000.aips-bus:ldb@020e0008: 0:imx_ldb_set_clock after: 475200000
[  503.429794] imx-ldb 2000000.aips-bus:ldb@020e0008: 0:imx_ldb_set_clock: now: 135771428 want: 138500000
[  503.429808] imx-ldb 2000000.aips-bus:ldb@020e0008: 0:imx_ldb_set_clock after: 135771428
[  503.429825] imx-ldb 2000000.aips-bus:ldb@020e0008: 1:imx_ldb_set_clock: LDB_CTRL: 0x000001fd
[  503.429834] imx-ldb 2000000.aips-bus:ldb@020e0008: 1:imx_ldb_set_clock: now: 475200000 want: 484750000
[  503.429848] imx-ldb 2000000.aips-bus:ldb@020e0008: 1:imx_ldb_set_clock after: 475200000
[  503.429856] imx-ldb 2000000.aips-bus:ldb@020e0008: 1:imx_ldb_set_clock: now: 135771428 want: 138500000
[  503.429867] imx-ldb 2000000.aips-bus:ldb@020e0008: 1:imx_ldb_set_clock after: 135771428
[  503.436828] imx-ldb 2000000.aips-bus:ldb@020e0008: imx_ldb_encoder_commit: LDB_CTRL: 0x000001fd
[  503.446974] it6251 2-005c: error -11 writing to edp addr 0x5
[  503.446985] dummy 2-005e: it6251.c:it6251_init:285 error -1 writing 255 to 5
[  503.456541] dummy 2-005e: error -11 writing to lvds addr 0x5
[  503.456553] dummy 2-005e: it6251.c:it6251_init:295 error -1 writing 255 to 5
[  503.550632] it6251 2-005c: System status: 0x3e
[  503.552120] it6251 2-005c: RPCLKCnt: 1520
[  503.553609] it6251 2-005c: Clock: 0x197
[  503.554356] it6251 2-005c: Ref Link State: 0x00
[  503.555111] it6251 2-005c: RPC Req: 0x24
[  503.556599] it6251 2-005c: hactive: 1920
[  503.558083] it6251 2-005c: vactive: 1080
...

**Comment by Jookia** [comment]
Ah, it might be a different bug then. Could you try this with my 4.19 buster stuff?

**Comment by gpsamu** [comment]
I can give it a whirl tomorrow.  Assuming I can easily revert back to 4.4 correct?  I read of the audio and power button light issues in 4.19.

**Comment by Jookia** [comment]
No, you can't revert back to 4.4 easily. You would need to make an SD card backup unfortunately.
Yeah there are still issues.

**Comment by gpsamu** [comment]
It's taken me a while, but I just confirmed the same issue with 4.19. I prepared a new SD card and ran through all of the upgrade steps including 5, which installs the 4.19 kernel.  I set the display power management as per the screenshot above and hit the same issue.  The display would not wake up with any keyboard/mouse activity, but I could still ping it.

**Comment by gpsamu** [comment]
Was just thinking.  My goal here was to have the display shut off entirely (backlight off) due to inactivity.  Is using the power management widget the correct way to achieve this.  Logically yes, but not sure if there is another way? Not really concerned with longevity as I think it's using LED backlights...

**Comment by Jookia** [comment]
I'm not too sure what the stack here is. I believe the correct way to do
this that the power management widget should use is DPMS.
